Dynatrace/dynatrace-operator
# Use correct tenantUUID

## Description
The `connectionInfo` we get from the cluster already contains the tenantUUID, and its the actual correct one, and not the one that we parse out of the apiURL, which caused problem if the apiURL was an alias of a tenant.

(Sidenote: most of the changes are related to rewriting the 22 months old tests 😅  )

## How can this be tested?
When using an apiURL that points to a tenant using an alias, setup cloudNativeFullStack (or application monitoring with CSI driver), check that the agents can communicate. (if alias is used in as the tenant, then the agents will complain about not being able to connect, this should not happen)

## Checklist
